### ISSUE TITLE
Filtering a category - different product positions in listing

### ISSUE THREAD

**Issue by ADDISON74** [OpenMage/magento-lts#1141]
### Preconditions (*)

1. Magento 1.9.4.5

### Steps to reproduce (*)

1. Go to a category with lots of products. In my case there are 70 products.
2. Filter the category by manufacturer attribute.
3. Repeat pressing F5 to reload the page and watch the products positions in listing.
4. Now navigate page by page. Remember products positions then go back.

### Expected result (*)

1. Products should have the same positions on pages no matter you list them on a page or navigating page by page in list.

### Actual result (*)

1. By reloading the page products have different positions. Navigating page by page with 12 results on a page a jacket listed on page 1 I found it listed again on page 5. 
2. As it is now Magento if you don't list all products in one page you can miss seeing products by navigating page by page. It is like a random listing.
3. This issue is not present when reloading a category without any filters. Products remain on the same position.

**Comment by ADDISON74** [comment]
For now I will close this issue. I failed to replicate it on a new system with the latest version of OpenMage v20.0.4 + sample data. I made a backup of the version with the problem and I will continue its investigation separately. In case I identify the source of the problem I will reopen the post.

**Comment by ADDISON74** [comment]
I was able to replicate this issue in a vanilla Magento 1.9.2.4.

- Category page + any filter applied => by reloading the page or navigate page by page the position of the products is different

- Search page (no need to apply a filter) => by reloading the page or navigate page by page the position of the products is different.

I migrated the database only into a OpenMage 20.0.4 and I was not able to replicate the issue. It is clear this issue was fixed in 1.9.3 - 1.9.4 but Magento team never told us about its existence.
